Math/NumberTheory/Moduli/PrimitiveRoot.hs

```diff
@@ -112,6 +119,13 @@ cyclicGroupToModulo = fromFactors . \case
   CGOddPrimePower p k       -> Coprimes.singleton (unPrime p) k
   CGDoubleOddPrimePower p k -> Coprimes.singleton 2 1 <> Coprimes.singleton (unPrime p) k
 
+-- | 'PrimitiveRoot m' is a type which is only inhabited by primitive roots of n.
+data PrimitiveRoot m =
+  PrimitiveRoot { unPrimitiveRoot :: MultMod m -- ^ Extract primitive root value.
+                , getGroup        :: CyclicGroup Natural -- ^ Get cyclic group structure.
+                }
```

Comment:
My personal preference is to put constructor on the same line with `data` and list fields indented by two spaces.